src/services/CertificateGenerationService.ts

```diff
@@ -807,6 +1039,183 @@ class CertificateGenerationService {
+                case "rowNumber":
+                  defectString += ` ${LOCATION_WELSH.ROW_NUMBER}: ${defect.additionalInformation.location.rowNumber}.`;
+                  break;
+                case "seatNumber":
+                  defectString += ` ${LOCATION_WELSH.SEAT_NUMBER}: ${defect.additionalInformation.location.seatNumber}.`;
+                  break;
+                case "axleNumber":
+                  defectString += ` ${LOCATION_WELSH.AXLE_NUMBER}: ${defect.additionalInformation.location.axleNumber}.`;
+                  break;
+                default:
+                  const welshLocation = this.convertLocationWelsh(
+                    defect.additionalInformation.location[location]
+                  );
+                  defectString += ` ${toUpperFirstLetter(welshLocation)}`;
+                  break;
+              }
+            }
+
+            if (index === array.length - 1) {
+              defectString += `.`;
+            }
+          }
+        );
+      }
+
+      if (defect.additionalInformation.notes) {
+        defectString += ` ${defect.additionalInformation.notes}`;
+      }
+      console.log(`Welsh Defect String Generated: ${defectString}`);
+      return defectString;
+    } else {
+      console.log(`ERROR: Unable to find a filtered defect`);
+      return null;
+    }
+  }
+
+  /**
+   * Returns welsh version of location
+   * @param locationToTranslate
+   */
+  public convertLocationWelsh(locationToTranslate: string) {
+    switch (locationToTranslate) {
+      case LOCATION_ENGLISH.FRONT:
+        return LOCATION_WELSH.FRONT;
+      case LOCATION_ENGLISH.REAR:
+        return LOCATION_WELSH.REAR;
+      case LOCATION_ENGLISH.UPPER:
+        return LOCATION_WELSH.UPPER;
+      case LOCATION_ENGLISH.LOWER:
+        return LOCATION_WELSH.LOWER;
+      case LOCATION_ENGLISH.NEARSIDE:
+        return LOCATION_WELSH.NEARSIDE;
+      case LOCATION_ENGLISH.OFFSIDE:
+        return LOCATION_WELSH.OFFSIDE;
+      case LOCATION_ENGLISH.CENTRE:
+        return LOCATION_WELSH.CENTRE;
+      case LOCATION_ENGLISH.INNER:
+        return LOCATION_WELSH.INNER;
+      case LOCATION_ENGLISH.OUTER:
+        return LOCATION_WELSH.OUTER;
+      default:
+        return locationToTranslate;
+    }
+  }
+
+  /**
+   * Returns filtered welsh defects
+   * @param filteredFlatDefects - the array of flattened defects
+   * @param vehicleType - the vehicle type from the test result
+   */
+  public filterFlatDefects(
+    filteredFlatDefects: IFlatDefect[],
+    vehicleType: string
+  ): IFlatDefect | null {
+    if (filteredFlatDefects.length === 0) {
+      return null;
+    } else if (filteredFlatDefects.length === 1) {
+      return filteredFlatDefects[0];
+    } else {
+      const filteredWelshDefectsOnVehicleType = filteredFlatDefects.filter(
+        (flatDefect: IFlatDefect) =>
+          flatDefect.forVehicleType!.includes(vehicleType)
+      );
+      return filteredWelshDefectsOnVehicleType[0];
+    }
```

Comment:
```suggestion
// return null stirahgt away
  if (!filteredFlatDefects.length) return null;

// return the defect otherwise search for it / return null
  if (filteredFlatDefects.length === 1) return filteredFlatDefects[0];
  return filteredFlatDefects.find(
    (x) => x.forVehicleType?.includes(vehicleType)
  ) || null;
```